alf/networks/preprocessor_networks.py

```diff
@@ -87,6 +88,7 @@ def _return_or_copy_preprocessor(preproc):
                 elif isinstance(preproc, InputPreprocessor):
                     preproc = preproc.copy()
                     self._input_preprocessor_modules.append(preproc)
+                    preproc = common.return_first(preproc)
```

Comment:
Seems using a decorator is not necessary, we can simply do:
```
elif isinstance(preproc, Network）：
   ...
   preproc = lambda x: preproc(x)[0]
```
Also at line 72, we use:
```
if not instance(preproc, Network):
 ...
else:
   return proproc.output_spec
```
Network always output_spec. We don't need to require preprocessor to calculate output spec given input spec.
And since we can use Network as preprocessor. It doesn't seem necessary to have class InputPreprocessor.

Reply:
Right, InputPreprocessor seems to be unnecessary now. Will try to remove it.

Reply:
Both the InputPreprocessor and decorator are removed now.